service/common/src/main/java/org/apache/polaris/service/exception/IcebergPersistenceExceptionMapper.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.exception;
+
+import jakarta.persistence.PersistenceException;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import jakarta.ws.rs.ext.Provider;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+
+
+@Provider
+public class IcebergPersistenceExceptionMapper
+    implements ExceptionMapper<PersistenceException> {
```

Comment:
I'm not sure creating a separate mapper just for `PersistenceException` is a good idea. `IcebergExceptionMapper` already deals with exception of multiple types. Could you integrate the new logic for `PersistenceException` there?

Reply:
I think it's a right thing to use a different class, instead of putting it into `IcebergExceptionMapper`. I even think we may pull non-iceberg exception out of the class `IcebergExceptionMapper`, whose name indicates that it handles iceberg exceptions, but not other exceptions. 

I'd suggest to rename it to `PersistenceExceptionMapper`.

Reply:
I also think it's reasonable to create a separate mapper for PersistenceException. In `IcebergExceptionMapper` we treat any "500" exceptions like "unhandled" and I would need to create yet another "if else" for PersistenceException which would make IcebergExceptionMapper even more complicated

Reply:
At the web layer exceptions may be wrapped in `RuntimeException` and other generic exception classes. There's usually logic to unwrap the root cause and process it depending on the specific type. Also, there's usually logic for logging interesting exceptions.

If more than one `ExceptionMapper` is used, provisions should be made to apply the same processing logic in all cases.

However, in this PR the exception processing logic is forked.